miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,769 @@
+        """Main brush usage time."""
+        return self._calcUsageTime(
+            self.main_brush_left, self.data["main_brush_life_level"]
+        )
+
+    @property
+    def main_brush_left(self) -> timedelta:
+        """How long until the main brush should be changed."""
+        return timedelta(minutes=self.data["main_brush_left_minutes"])
+
+    @property
+    def side_brush(self) -> timedelta:
+        """Main brush usage time."""
+        return self._calcUsageTime(
+            self.side_brush_left, self.data["side_brushes_life_level"]
+        )
+
+    @property
+    def side_brush_left(self) -> timedelta:
+        """How long until the side brushes should be changed."""
+        return timedelta(minutes=self.data["side_brushes_left_minutes"])
+
+    @property
+    def sensor_dirty(self) -> timedelta:
+        """Return time since last sensor clean."""
+        return self._calcUsageTime(
+            self.sensor_dirty_left, self.data["sensor_dirty_remaning_level"]
+        )
+
+    @property
+    def sensor_dirty_left(self) -> timedelta:
+        """How long until the sensors should be cleaned."""
+        return timedelta(minutes=self.data["sensor_dirty_time_left_minutes"])
+
+
+class RoidmiVacuumMiot(MiotDevice):
+    """Interface for Vacuum Eve Plus (roidmi.vacuum.v60)"""
+
+    mapping = _MAPPING
+
+    @command()
+    def status(self) -> RoidmiVacuumStatus:
+        """State of the vacuum."""
+        return RoidmiVacuumStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                # max_properties limmit to 10 to avoid "Checksum error" messages from the device.
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command()
+    def consumable_status(self) -> RoidmiConsumableStatus:
+        """Return information about consumables."""
+        return RoidmiConsumableStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                # max_properties limmit to 10 to avoid "Checksum error" messages from the device.
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command()
+    def cleaning_summary(self) -> RoidmiCleaningSummary:
+        """Return information about cleaning runs."""
+        return RoidmiCleaningSummary(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                # max_properties limmit to 10 to avoid "Checksum error" messages from the device.
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command()
+    def start(self) -> None:
+        """Start cleaning."""
+        return self.call_action("start")
+
+    @command(click.argument("roomstr", type=str))
+    def start_room_sweep_unknown(self, roomstr: str) -> None:
+        """Start room cleaning.
+
+        FIXME: the syntax of room_sweep is unknown
```

Comment:
Considering `roomstr` is not used at all, does something happen if this command gets executed?

If something happens, I'd move the FIXME to be `_LOGGER.warning()` to let users know about it (maybe someone finds out what the string should be) and otherwise remove the command decorator (so that it's not shown in the mioocli but left here just for a reference).

Reply:
The problem is that I have not figured out how to start an "action" with an argument.
I will comment out all the functions that are calling an action with an argument.